airflow/example_dags/example_datasets.py

```diff
@@ -41,12 +41,11 @@
 DAG example_dataset_dag9 should fail its only task and never trigger example_dataset_dag10_req_dag9
 
 """
```

Comment:
Took the liberty to update this DAG to use TaskFlow API generally and conform the `start_date` declaration we've been using for example DAGs.

Reply:
Yeah I saw similar behaviour happening more frequently since we [updated to the newer mypy version](https://github.com/apache/airflow/pull/25088) - there are more cases when "small set of files changed" produced different result than "big set of files". I will investigate if we can make MyPy more "greedy" in terms of considering bigger set of files and searching "deeper" if only few files changed  - because ultimately I think  this is what's happening, that MyPy is not getting all the information neded to arrive to the same conclusion about parsing. 

I will look a bit deeer on how MyPy makes it's decisions and maybe we can improve some of this "partial changes mypy" experience.